lib/gitlib.sh

```diff
@@ -20,8 +20,13 @@
 ###############################################################################
 # CONSTANTS
 ###############################################################################
+
 GITHUB_TOKEN=${FLAGS_github_token:-$GITHUB_TOKEN}
-[[ -n $GITHUB_TOKEN ]] && GITHUB_TOKEN_FLAG=("-u" "$GITHUB_TOKEN:x-oauth-basic")
+
+# This is a safety measure to trim extraneous leading/trailing whitespace from
+# the OAuth token provided. Otherwise, authentication to GitHub will fail here.
+GITHUB_TOKEN="$( echo "$GITHUB_TOKEN" | tr -d '[:space:]' )"
+[[ -n "$GITHUB_TOKEN" ]] && GITHUB_TOKEN_FLAG=("-u" "${GITHUB_TOKEN}:x-oauth-basic")
```

Comment:
consider always putting braces around variables :-)

Reply:
@BenTheElder -- heh, yeah. At this point, I'm just trying to stop the bleeding :sweat_smile: